### ISSUE TITLE
CantusDb json-ld expands to include invalid URI's

### ISSUE THREAD

**Issue by dchiller** [malajvan/linkedmusic-datalake#9]
When I expand the current CantusDB jsonld, I get some (I think) invalid URI's. 

For example, the first chant has unexpanded jsonld: 

```json
    {
        "volpiano": "1---kl--lk--lm--m--m---3---k--l--kh--k--l--k7---3",
        "@context": "https://raw.githubusercontent.com/malajvan/linkedmusic-datalake/main/cantusdb/jsonld/context.jsonld",
        "@id": "chant:643187",
        "@type": "wd:Q23072435",
        "database": "cantusdb:",
        "P86": "wd:Q4233718",
        "P1922": "Benedixisti*",
        "P136": "wd:Introit verse",
        "Q731978": "wd:None",
        "Q4484726": "C",
        "source": "src:638308"
    },
```

When I expand this using the context, though, I get:

```json
{
  "@id": "https://cantusdatabase.org/chant/643187",
  "@type": [
    "http://www.wikidata.org/entity/Q23072435"
  ],
  "http://www.wikidata.org/prop/direct/P136": [
    {
      "@id": "http://www.wikidata.org/entity/Introit verse"
    }
  ],
  "http://www.wikidata.org/prop/direct/P1922": [
    {
      "@value": "Benedixisti*"
    }
  ],
  "http://www.wikidata.org/prop/direct/P86": [
    {
      "@id": "http://www.wikidata.org/entity/Q4233718"
    }
  ],
  "http://www.wikidata.org/entity/Q4484726": [
    {
      "@value": "C"
    }
  ],
  "http://www.wikidata.org/entity/Q731978": [
    {
      "@id": "http://www.wikidata.org/entity/None"
    }
  ],
  "https://schema.org/Dataset": [
    {
      "@id": "https://cantusdatabase.org/"
    }
  ],
  "https://cantusdatabase.org/source/": [
    {
      "@id": "https://cantusdatabase.org/source/638308"
    }
  ],
  "https://web.mit.edu/music21/doc/moduleReference/moduleVolpiano.html": [
    {
      "@value": "1---kl--lk--lm--m--m---3---k--l--kh--k--l--k7---3"
    }
  ]
}
```

But I think 
> http://www.wikidata.org/entity/None
and 
> http://www.wikidata.org/entity/Introit verse
are invalid.

**Comment by malajvan** [comment]
Hi! I am aware of the issue. For the None, definitely should handle that and result to null rather than an iri. 
For the Introit, it's because the rest of the columns are reconciled except for a few of them (take Introit as an example, but now that I manually looked it up it does exist at Q224377) and so I was unsure how I should handle them: 
1. Resolve to None
2. make them values instead of ids and keep the string 'Introit'

I think 2 is the best option but I don't know if it will mess with Solr if some P136(genre) is an @id (IRI) and some are @values (string)

Let me know what you think

Edit: I now realize that I should actually reconcile those Introit manually ie use another link (not wikidata). Let me know what you think

**Comment by malajvan** [comment]
From today's meetings, I'll generate 2 different json-ld files: One linking the non reconciled to an external link, and another keep them as text

**Comment by malajvan** [comment]
@dchiller I fixed this and generated 2 compact jsonld for cantusdb, 1 for replacing with links(ids) and 1 for replacing with string(values).
you can find them here:
[link_compact.jsonld](https://github.com/malajvan/linkedmusic-datalake/blob/main/cantusdb/jsonld/link_compact.jsonld)
[val_compact.jsonld](https://github.com/malajvan/linkedmusic-datalake/blob/main/cantusdb/jsonld/val_compact.jsonld)
I will now do handle errors in simssadb (keep as values).

**Comment by dchiller** [comment]
TY @malajvan

**Comment by annamorphism** [comment]
late to this discussion but note that Q224377 is just vanilla "Introit"--CantusDB makes a distinction between this and the (related) "Introit Verse".

**Comment by malajvan** [comment]
@annamorphism I also noticed this and reconciled only the "Introit". For "Introit verse" I left them as strings (same as other "verses"). You can see an example here: [line 619](https://github.com/malajvan/linkedmusic-datalake/blob/main/cantusdb/jsonld/compact.jsonld#L619C12-L619C12).
Let me know if this is the correct approach

**Comment by annamorphism** [comment]
yes I think this is correct! (at least, as long as it doesn't matter that these two somewhat-related-things have different kinds of properties, but that is the point of this discussion!)
